ts/Stock/Indicators/Stochastic/StochasticIndicator.ts

```diff
@@ -205,11 +209,28 @@ class StochasticIndicator extends SMAIndicator {
             HL = extremes[1] - LL;
             K = CL / HL * 100;
 
+            if (isNaN(K) && constantValues) {
+                j++;
+                continue;
+            } else if (constantValues && !isNaN(K)) {
+                constantValues = false;
+            }
+
             xData.push(xVal[i]);
-            yData.push([K, null]);
+
+            if (isNaN(K)) {
```

Comment:
Could you please add a comment with an explanation of what actually happens inside the if-else? The one we discussed a few weeks ago: imho the logic is not so self-explanatory